### ISSUE TITLE
Prettier formatting stops working for documents with tables

### ISSUE THREAD

**Issue by oliversturm** [neilsustc/vscode-markdown#317]
#### What is the problem?

I have auto-formatting on save switched on. As a test case, I edit a markdown file and I include extra spaces between words or at the ends of lines, and extra empty lines. When I save the document, this extra whitespace is removed.

As soon as I include a table in the markdown, this functionality breaks. The table is formatted correctly, with extra dashes and spaces inserted. But the document formatting functionality described above stops working.

I have *Enable GFM Table Formatter* active in the settings. However, I have tried switching it off and the problem remains the same.

I have also tried to disable the *Markdown All in One* plugin, and then document formatting works correctly with the table in the document. Of course table formatting doesn't work in that case.

#### How can I reproduce it?

Here's a small test file I used: https://gist.github.com/oliversturm/d763c8c4fa61405db3f570c686fb7866

#### Is there any error message in the console?

No. I have also checked various other log outputs in VS Code, but I didn't find any relevant messages.

My VS Code version is 1.28.2 and I'm running on Linux. My plugin version is 1.7.0. However, I have noticed this issue for a while now, so past versions of the last few months would probably show the same results.

**Comment by oliversturm** [comment]
In further tests I just realized that the document formatting functionality I'm referring to is actually provided by [Prettier](https://github.com/prettier/prettier-vscode). So I guess this is a case of incompatibility between these plugins.

**Comment by neilsustc** [comment]
Thanks for the detailed description. The problem is vscode implicitly only allows one "formatter" per language. Then this extension('s formatter) and that of Prettier cannot co-exist.

Related Microsoft/vscode#41882.

---

Is there an option of Prettier to format GFM table? If it can do this, you can disable this extension's formatter with `markdown.extension.tableFormatter.enabled`

**Comment by oliversturm** [comment]
> Is there an option of Prettier to format GFM table?

That is an interesting question. I didn't think there was, because table formatting was one of the reasons for me to start using this plugin at first, and I've been using Prettier much longer. However, now I tested that explicitly and it actually does appear that Prettier performs table formatting as well. So that's good news for me.

I was hesitant about your suggestion to disable `markdown.extension.tableFormatter.enabled` since I tried that earlier. However I tried it again now and I found that the setting didn't take hold before because I did not reload the window. I can now confirm that Prettier and this plugin can work side by side correctly as long as GFM table formatting is disabled in the latter.

Thanks for your help!
